aten/src/ATen/native/MathBitsFallback.cpp

```diff
@@ -0,0 +1,119 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <torch/library.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/native/UnaryOps.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+
+struct MathOpFallback {
+  MathOpFallback(DispatchKey key_, string op_name_) : key(key_), op_name(op_name_) {}
+  virtual bool is_bit_set(c10::TensorImpl&) = 0;
```

Comment:
There isn't really any reason for this to take a mutable reference. Also the TensorImpl here is weirdly asymmetric with the Tensor everywhere else; maybe there is a good reason for this but it is not immediately evident.